monai/utils/misc.py

```diff
@@ -352,6 +356,91 @@ def copy_to_device(
     return obj
 
 
```

Comment:
have you checked that the methods are equivalent as the previous ignite APIs? https://github.com/pytorch/ignite/tree/master/ignite/distributed looks like the get device logic is different

Reply:
The ignite `device` API is: https://github.com/pytorch/ignite/blob/master/ignite/distributed/comp_models/native.py#L283
I think if user set `backend = GLOO`, the device should be `CPU` instead of `CUDA:x`, what do you think?

Thanks.

Reply:
ok, I haven't looked into the details I think we can update it later if needed

Reply:
sure, sounds good!
Thanks.